File: merge-strings-alternately/merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    var result = ""
    if (word1.length > word2.length) {
        for (let i = 0; i < word2.length; i++) {
            result += word1[i]
            result += word2[i]
        }
        result += word1.slice(word2.length)
        return result
    } else if (word2.length > word1.length) {
        for (let i = 0; i < word1.length; i++) {
            result += word1[i]
            result += word2[i]
        }
        result += word2.slice(word1.length)
        return result
    } else {
        for (let i = 0; i < word1.length; i++) {
            result += word1[i]
            result += word2[i]
        }
        return result
    }
};

File: merge-strings-alternately/merge-strings-alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        result = ""
        if len(word1) > len(word2):
            for i in range(len(word2)):
                result += word1[i]
                result += word2[i]
            result += word1[len(word2):]
            return result
        elif len(word2) > len(word1):
            for i in range(len(word1)):
                result += word1[i]
                result += word2[i]
            result += word2[len(word1):]
            return result
        else:
            for i in range(len(word1)):
                result += word1[i]
                result += word2[i]
            return result